lib/devise/rails/deprecated_constant_accessor.rb

```diff
@@ -28,7 +28,7 @@ def const_missing(missing_const_name)
 
           def deprecate_constant(const_name, new_constant, message: nil, deprecator: Devise.deprecator)
             class_variable_set(:@@_deprecated_constants, {}) unless class_variable_defined?(:@@_deprecated_constants)
-            class_variable_get(:@@_deprecated_constants)[const_name.to_s] = { new: new_constant, message: message, deprecator: deprecator }
+            class_variable_get(:@@_deprecated_constants)[const_name.to_s] = { new: new_constant, message: message, deprecator: Devise.deprecator }
```

Comment:
Not sure about this fix

The current implementation should work, in fact the deprecation warning is not present on main branch.

Please try 

```
gem 'devise', github: 'heartcombo/devise'
```

in your application Gemfile

Fixed via f8f035155b9e00e6c6f702f81955bf89dfdb4a60
Ref: heartcombo/devise#5628

Reply:
Yes, you're absolutely right. It worked without my fix just due to unreleased mainline fixes. 🤦‍♂️